drivers/platform/maxim/maxim_gpio.c

```diff
@@ -390,60 +415,38 @@ static int32_t max_gpio_irq_set_trigger_level(struct irq_ctrl_desc *desc,
 	struct gpio_desc *g_desc;
 	mxc_gpio_cfg_t *max_gpio_cfg;
 	mxc_gpio_regs_t *gpio_regs;
-	uint32_t is_enabled;
+	uint32_t mask;
 
-	if (!desc || !desc->extra || irq_id > N_PINS)
+	if (!desc || !desc->extra || irq_id >= N_PINS)
 		return -EINVAL;
 
 	g_desc = desc->extra;
 	max_gpio_cfg = g_desc->extra;
-	gpio_regs = max_gpio_cfg->port;
-	is_enabled = gpio_regs->int_en & BIT(irq_id);
-
-	/** Disable interrupts for pin desc->number */
-	gpio_regs->int_en &= ~(BIT(irq_id));
-	/** Clear pending interrupts for pin desc->number */
-	gpio_regs->int_clr |= BIT(irq_id);
-	/** Disable dual edge interrupts for pin desc->number */
-	gpio_regs->int_dual_edge &= ~BIT(irq_id);
+	mask = max_gpio_cfg->mask;
+	max_gpio_cfg->mask = BIT(irq_id);
```

Comment:
What does the mask represent here? Does it contain the enabled pins in a port? And if so, should this operation be `max_gpio_cfg->mask &= BIT(irq_id);` to also check that the bit is active?

Reply:
This struct (max_gpio_cfg) is used to specify a GPIO port configuration (so the mask field holds the pins on which the configuration should be done). However, the struct is not used to store the entire port state, meaning that you can't check if a pin is enabled by reading the mask field. For that you would have to read the en0 register. Also, I'm not sure the check is necessary because I imagine the interrupt setting sequence as following:

`irq_register_callback()
irq_trigger_level_set()
irq_enable()
`
The `irq_register_callback` call configures the pin as input (which means that it's enabled).

Reply:
Also, the user has to provide a `gpio_desc` as extra for the irq descriptor (for the pin on which the interrupt would be set), so I assume that the GPIO pin has been configured already.